docs/tekton-bundle-contracts.md

```diff
@@ -0,0 +1,38 @@
+<!--
+---
+linkTitle: "Tekton Bundle Contracts"
+weight: 8
+---
+-->
+# Tekton Bundle Contract
+
+When using a Tekton Bundle in a task or pipeline reference, the OCI artifact backing the
+bundle must adhere to the following contract.
+
+## Contract
+
+Only Tekton CRDs (eg, `Task` or `Pipeline`) may reside in a Tekton Bundle used as a Tekton
+bundle reference.
+
+Each layer of the image must map 1:1 with a single Tekton resource.
+
+Each layer must contain the following annotations:
+
+- `org.opencontainers.image.title` => `ObjectMeta.Name` of the resource
+- `cdf.tekton.image.kind` => `TypeMeta.Kind` of the resource, all lowercased (eg, `task`)
+- `cdf.tekton.image.apiVersion` => `TypeMeta.APIVersion` of the resource (eg 
```

Comment:
should we use `dev.tekton.*` instead here ? (following the opencontainers.org pattern)

Reply:
+1.

Reply:
This does make the translation a bit more icky because we would have to translate the `TypeMeta` of the desired object to the `dev.tekton.*` form. I'm not against this but I am curious if we think the gain is worth the added logic?

Reply:
what's the difference between `cdf.tekton.*` and `dev.tekton.*` ? :thinking:

Reply:
I think dev.tekton* would be more accurate. Do we already have people using the cdf.tekton form somehow?

Reply:
I will raise my hand and apologize here. I misunderstood the original intent. I thought you meant that the "value" inside of this annotation should be `dev.tekton` rather than changing the annotation name. Yeah I am game to change the annotation name 😄